### ISSUE TITLE
Feedback - Talent Guide doesn't account for self options

### ISSUE THREAD

**Issue by Starlynk1** [RestedXP/RXPGuides#199]
Playing a Paladin and the talent guide doesn't normally have you pick up the Consecration talent. I like having this as soon as I can get it as I find it helpful. The problem is that if you choose a talent outside of the Talent Guide parameters, it doesn't account for that choice.

Would like to see the talent guide adjust itself so that choices outside of the guides recommended cause the recommendations to adjust what level to get the other talents. Screenshot shows that it says to get the next level of Retribution talents, but I have not put in 5 points yet for the previous tier. I know I need to mentally adjust the numbers, but having the addon do this would be great, if possible.

![12df72d5c073828bd0dd82c62594de40](https://github.com/RestedXP/RXPGuides/assets/91444596/26a9aebc-7d5c-44fd-bc36-a8ecd0aa16f9)

**Comment by SabreValkyrn** [comment]
This was something I considered and wanted to do when speccing out this process.

However, realistically it's rather difficult to handle all the permutations and calculations with user choice/variations.

The best I've thought of is blocking auto-apply if it detects a variance in choices, but still keep the level ranges visible so more advanced users can see what the guide recommends.

**Comment by Starlynk1** [comment]
I'm not sure how the guide is written for talents and how it determines what is going where. My guess is that it first looks at race/class, then softcore or hardcore option. Depending on that, it follows a pre-set list of talents and ranks according to character level. Then adds those to the talent window and assigns that way. If that is the case or its something close, then my thought was to have an AdjustedTalentLevel to account for mismatches in the talent levels. So the talent guide doesn't change, it just adjusts each talent down the list further but continues along the same path of recommendations by level.

Basically, any time the CharLevel doesn't match the expected level of the talent it wants to use, then it would increment the the AdjTalentlevel to the next level (or more) to line up the talent acquisitions again. 

I think doing it this way, there wouldn't be any changes need to the existing talent list by race/class, just adjustments to the suggested levels based on charlevel difference.

Could also add a popup to disable the talent guide if they deviate X amount of times.

**Comment by Starlynk1** [comment]
Just looked at the Hunter talent guide and it appears there are no set level assignments to each talent. Which I think may make it easier since its not hardcoded to a level. Same idea, but instead of checking level, check to see what was the last talent/rank that had been acquired in the list and what has not been acquired yet. Then change the levels based on what the next char level will be. Hope that made sense
